code/modules/overmap/ships/computers/ship.dm

```diff
@@ -8,27 +8,37 @@ somewhere on that shuttle. Subtypes of these can be then used to perform ship ov
 	var/list/viewers // Weakrefs to mobs in direct-view mode.
 	var/extra_view = 0 // how much the view is increased by when the mob is in overmap mode.
 
+/obj/machinery/computer/ship/New()
+	. = ..()
+	var/list/L = list()
+	for(var/type in whitelisted_types)
+		L |= typesof(type)
+	for(var/type in blacklisted_types)
+		L -= typesof(type)
+	whitelisted_types = L
```

Comment:
Sure uses a lot of memory on every ship computer if they have whitelisted types.